packages/lesswrong/lib/sql/TableIndex.ts

```diff
@@ -18,18 +21,20 @@ class TableIndex {
     if (options?.partialFilterExpression && !options.name) {
       this.name += "_filtered";
     }
+    if (options?.collation) {
+      this.collationType = getCollationType(options.collation);
+      if (this.collationType === "case-insensitive") {
+        this.name += "_ci";
+      }
+    }
   }
 
   getFields() {
     return this.fields;
   }
 
-  // TODO: This is lossy and may lead to collisions
   private getSanitizedFieldNames() {
-    return this.fields.map((field) => {
-      const index = field.indexOf(".");
-      return index >= 0 ? field.slice(0, index) : field;
-    });
+    return this.fields.map((field) => field.replace(/\./g, "__"));
```

Comment:
Will this lead to a bunch of duplicate indexes?

Reply:
Outside `Users` there's only one index in the entire codebase that uses this code path, so only that one will be duplicated. I'm going to just remove it manually ~a week after we do the migration, otherwise it will get recreated any time someone connects to the DB on a branch that doesn't have these changes.